Clarify USM environment variable detection limitations (#32626)
DataDog/documentation: content/en/universal_service_monitoring/setup.md

- Datadog Agent is installed alongside your service. Installing a tracing library is _not_ required.
- The `env` tag for [Unified Service Tagging][1] has been applied to your deployment. The `service` and `version` tags are optional.

## How USM detects service names

<div class="alert alert-warning">
<strong>Important:</strong> Universal Service Monitoring detects service names from environment variables that exist when a process starts. USM reads these values from the operating system (from <code>/proc/PID/environ</code> on Linux or via system APIs on Windows).
</div>

**Environment variables USM recognizes:**
- `DD_SERVICE` - Explicitly sets the service name
- `DD_TAGS` - Can include `service:name` tag
- `DD_ENV` - Sets the environment tag
- `DD_VERSION` - Sets the version tag

**Key limitation:**

> Setting environment variables programmatically **inside your application code** (such as `System.setProperty("dd.service", "my-service")` in Java or `Environment.SetEnvironmentVariable("DD_SERVICE", "my-service")` in .NET) will **not** be detected by USM, even though these values work for APM tracing instrumentation.

Reviewer: ```suggestion
Environment variables programmatically set **inside your application code** (such as `System.setProperty("dd.service", "my-service")` in Java or `Environment.SetEnvironmentVariable("DD_SERVICE", "my-service")` in .NET) are **not** detected by USM, even though these values work for APM tracing instrumentation.
```
